src/org/traccar/protocol/Tk103FrameDecoder.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2017 Valerii Vyshniak (val@val.one)
+ * Copyright 2017 Anton Tananaev (anton@traccar.org)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.traccar.protocol;
+
+import org.jboss.netty.buffer.ChannelBuffer;
+import org.jboss.netty.channel.Channel;
+import org.jboss.netty.channel.ChannelHandlerContext;
+import org.jboss.netty.handler.codec.frame.FrameDecoder;
+
+public class Tk103FrameDecoder extends FrameDecoder {
+
+    private static final int FRAME_MIN_SIZE = 2;
+    private static final int FRAME_MAX_SIZE = 1024;
+    private static final byte FRAME_START_SYMBOL = (byte) '(';
+    private static final byte FRAME_END_SYMBOL = (byte) ')';
+    private static final byte FRAME_FREE_TEXT_BLOCK_SYMBOL = (byte) '$';
+
+    @Override
+    protected Object decode(
+            ChannelHandlerContext ctx,
+            Channel channel,
+            ChannelBuffer buf) throws Exception {
+
+        if (buf.readableBytes() < FRAME_MIN_SIZE) {
+            return null;
+        }
+
+        int indexStart = buf.indexOf(buf.readerIndex(), buf.writerIndex(), FRAME_START_SYMBOL);
+        if (indexStart == -1) {
+            buf.clear();
+            return null;
+        }
+        indexStart++;
+
+        int index, cnt;
```

Comment:
Please use proper names for variables. In this case probably `count` instead of `cnt`.

Reply:
done